### PR TITLE
feat: Add hemi-sphere area algorithm 

### PR DESCRIPTION
**Pull request by HarshilShah1804**
#### Description of Change
Added the area of hemi-sphere algorithm with test to math/area.cpp 

#### Checklist

- [x] Added description of change
- [x] Added tests and example, test must pass
- [x] Added documentation so that the program is self-explanatory and educational - [Doxygen guidelines](https://www.doxygen.nl/manual/docblocks.html)
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes:

### CODE REVIEW

File: math/area.cpp [hunk 1]
```diff
@@ -109,6 +109,17 @@ template <typename T>
 T cylinder_surface_area(T radius, T height) {
     return 2 * M_PI * radius * height + 2 * M_PI * pow(radius, 2);
 }
+
+/**
+ * @brief surface area of a [hemi-sphere](https://en.wikipedia.org/wiki/Surface_area) ( 3 *
+ * pi * r^2)
+ * @param radius is the radius of the hemi-sphere
```
Comment (realstealthninja) [inline math/area.cpp:116]:
```suggestion
 * @param radius is the radius of the hemi-sphere
 * @tparam T datatype of radius
```

**Review by realstealthninja — CHANGES_REQUESTED** [review]

**Comment by realstealthninja** [comment]
hey thank you for your contribution! please let me know if this is for hacktoberfest!

**Comment by HarshilShah1804** [comment]
hey thank you for review. 

Yes, this was for hacktoberfest.

**Review by HarshilShah1804 — COMMENTED** [review]
I have committed your suggestion.

**Review by realstealthninja — APPROVED** [review]
Wonderfully done! Thank you so much for this contribution ill mark this as hacktoberfest-accepted. I wish you the best of luck!

**Pull request merged**